tests/product/timing_test_decorator.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import sys
+
+from re import match
+from time import time
+
+
+def _initialize_module_logger(module_name):
+    logger = logging.getLogger(module_name)
+    logger.setLevel(logging.INFO)
+
+    message_format = '%(name)s - %(message)s'
+    formatter = logging.Formatter(message_format)
+
+    console_handler = logging.StreamHandler(sys.stdout)
+    console_handler.setLevel(logging.INFO)
+    console_handler.setFormatter(formatter)
+
+    if not len(logger.handlers):
+        logger.addHandler(console_handler)
+    return logger
+
+
+def _name_starts_with_test(name):
+    pattern = '^test.*$'
+    return match(pattern, name)
+
+
+# Logging is disabled by default for product tests
+# Logging can be enabled using this module by uncommenting the lines:
+# logging.disable(logging.NOTSET) in the function below
+def log_function_time(module_name="root"):
+    """
+    Args:
+        module_name: The module of the function that is being timed
+         By default, this is set to 'root' to use the root logger.
+         It is recommended that __name__ be passed into this function
+         for the most accurate logging.
+
+    Usage:
+         @log_function_time(__name__)
+         def my_function(params):
+
+    Returns: Prints the execution time of the decorated function to the
+    log file(s). It will always log as info the execution time to the
+    info log file. If the execution time exceeds 10 minutes, it will also
+    log it to the error log file.
+    """
+    def name_wrapper(function):
+        def time_wrapper(*args, **kwargs):
+            # logging.disable(logging.NOTSET)
+            logger = _initialize_module_logger(module_name)
+            function_name = function.__name__
+            function_is_a_test = _name_starts_with_test(function_name)
+            if function_is_a_test:
+                logger.info("%s started", function_name)
+            start_time = time()
+            logging.disable(logging.CRITICAL)
+
+            return_value = function(*args, **kwargs)
+
+            # logging.disable(logging.NOTSET)
+            elapsed_time = time() - start_time
+            travis_output_time_limit = 600
+            message_level = logging.ERROR if elapsed_time >= travis_output_time_limit \
+                else logging.INFO
+            if function_is_a_test:
```

Comment:
I found the distinction helpful when I was working with a lot of tests, but since this decorator will likely be used infrequently and on single tests, I think you're right. I'll go ahead and get rid of the distinction.